mar/graph/validator/library.hoon

```diff
@@ -0,0 +1,101 @@
+/-  *post, met=metadata-store, graph=graph-store
+=>
+|%
+++  is-title-valid
+  |=  [title=tape]
+  ^-  ?
+  %.y
+++  is-isbn-valid  :: we could do https://github.com/xlcnd/isbnlib/blob/41f59c74a69a2675c3f135431e9785f9ae502a7e/isbnlib/_core.py#L52
+  |=  [isbn=tape]
+  ^-  ?
+  =/  len  (lent isbn)
+  ?>  ?|(=(len 10) =(len 13))
+  %.y
+--
+|_  i=indexed-post
+++  grow
+  |%
+  ++  noun  i
+  ::
+  ++  graph-indexed-post
+    ^-  indexed-post
+    ?+    index.p.i  !!
+    ::  top level node: book
+    ::  structural node with no content
+    ::
+        [@ ~]
+      ~|  "top level book node should be empty!"
+      ?>  ?=(~ contents.p.i)
+      i
+    ::  metadata revision container
+    ::  structural node with no content
+    ::
+        [@ %meta ~]
+      ~|  "metadata revision container should be empty!"
+      ?>  ?=(~ contents.p.i)
+      i
+    ::  single metadata revision
+    ::  content node. first %text element is treated as title,
+    ::  second is treated as isbn
+    ::
+        [@ %meta @ ~]
+      =/  contents  contents.p.i
+      ?>  ?=([[%text *] [%text *] ~] contents.p.i)
+      =/  title  (trip +.i.contents.p.i)
+      ?>  (is-title-valid title)
```

Comment:
Why does this need a tape rather than a cord? Same question for ISBN

Reply:
Hmmm. tbh I didn't put too much thought into it so its probably just a not so good decision. Looking at `cord`, it supports UTF-8 and `tape` doesn't so thats one more argument against it. I think the main reason was because I didn't know/forgot how to get the length of a `cord`. For the length check for the isbn. Once I know how to do that I'll change both to cords.

Reply:
ok thinking about it more, isbn should only ever be numbers and dashes so it makes sense for it to remain a tape. not title, however.

Reply:
https://github.com/ynx0/library/pull/3/commits/f86bedbaefebcc85541a15b7c411499369e4c680